components/selection/selection-summary.js

```diff
@@ -35,25 +35,53 @@ class Summary extends LocalizeCoreElement(SelectionObserverMixin(LitElement)) {
 	}
 
 	render() {
+		if (!this._summary) {
+			return nothing;
+		}
+		return html`
+			<p class="d2l-body-compact">${this._summary}</p>
+		`;
+	}
+
+	willUpdate(changedProperties) {
+		if (changedProperties.has('_provider') || changedProperties.has('selectionInfo')) {
+			this._updateSelectSummary();
+		}
+	}
+
+	_updateSelectSummary() {
 		if (this._provider && this._provider.selectionSingle) return;
 
 		let count;
-		let summary;
 		if (this._provider && this._provider.selectionCountOverride !== undefined) {
 			count = this._provider.selectionCountOverride;
-			summary = (this._provider.selectionCountOverride === 0 && this.noSelectionText ?
-				this.noSelectionText : this.localize('components.selection.selected', 'count', count));
+			this._summary = this._provider.selectionCountOverride === 0 && this.noSelectionText ?
+				this.noSelectionText : this.localize('components.selection.selected', 'count', count);
 		} else {
-			count = (this.selectionInfo.state === SelectionInfo.states.allPages ?
-				this._provider.itemCount : this.selectionInfo.keys.length);
+			let includePlus = false;
+			if (this._provider && this._provider._getLazyLoadItems) {
```

Comment:
If we have any more changes to make, I wouldn't mind if there was a comment for other devs who might not be as familiar... something to the effect of "If lazy loading items is supported (ex. d2l-list) then check the keys to determine if the plus sign should be included. If lazy loading is not supported (ex. d2l-table-wrapper), then skip this.". It might not be particularly evident at first glance that `_getLazyLoadItems` is part of the provider API... so a callout might be helpful.

Reply:
Good call, I'll add that comment 👍🏻